### ISSUE TITLE
About detach in D_step

### ISSUE THREAD

**Issue by wang975679801** [ARASC/CFGAN#2]
How to train the Discriminator by realData_result.detach().numpy?

fakeData_result=D(fakeData_ZP,useInfo_batch) 
realData_result=D(realData,useInfo_batch) 
d_loss = -np.mean(np.log(realData_result.detach().numpy()+10e-5) + 
                              np.log(1. - fakeData_result.detach().numpy()+10e-5)) + 0*regularization(fakeData_ZP,realData_zp)
